src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockGoogleDataprocDAO.scala

```diff
@@ -27,4 +35,34 @@ class MockGoogleDataprocDAO extends DataprocDAO {
       clusters.remove(clusterName)
     Future(())
   }
+
+  override def updateFirewallRule(googleProject: GoogleProject): Future[Unit] = {
+    Future.successful(
```

Comment:
Nitpick, but can these methods be written like this:
```
if (!firewallRules.contains(googleProject)) {
  firewallRules += googleProject -> dataprocConfig.clusterFirewallRuleName
}
Future.successful(())
```
I think the code is equivalent, but IMO it makes it more clear that updating the mutable collections is done synchronously. `ArrayBuffer` is not a concurrent data structure, so I'm paranoid about modifying it inside a `Future`.